aten/src/ATen/core/Tensor.h

```diff
@@ -619,6 +619,8 @@ class CAFFE2_API Tensor {
   Tensor dequantize() const;
   double q_scale() const;
   int64_t q_zero_point() const;
+  Tensor q_scales() const;
```

Comment:
I feel like this bloats the current tensor implementation. Can we somehow implement polymorphic returns? Maybe through covariant return types or pointer returns, while keeping the name of the getter the same (I mean `q_scale`)?

Reply:
this is planed but it has dependencies on TorchBind which is still in experimental, so we are not committed to do this by the release